src/category_theory/limits/fubini.lean

```diff
@@ -0,0 +1,243 @@
+          dsimp at this,
+          simp only [category.id_comp, category_theory.functor.map_id, nat_trans.id_app] at this,
+          exact this,
+        end } },
+    naturality' := λ j j' f, (Q j').hom_ext
+    begin
+      dsimp,
+      intro k,
+      simp only [limits.cone_morphism.w, limits.cones.postcompose_obj_π, limits.is_limit.fac_assoc,
+        limits.is_limit.fac, nat_trans.comp_app, category.id_comp, category.assoc],
+      have := @nat_trans.naturality _ _ _ _ _ _ c.π (j, k) (j', k) (f, 𝟙 k),
+      dsimp at this,
+      simp only [category.id_comp, category.comp_id,
+        category_theory.functor.map_id, nat_trans.id_app] at this,
+      exact this,
+    end, } }.
+
+/--
+`cone_of_cone_uncurry Q c` is a limit cone when `c` is a limit cone.`
+-/
+def cone_of_cone_uncurry_is_limit
+  {D : diagram_of_cones F} (Q : Π j, is_limit (D.obj j))
+  {c : cone (uncurry.obj F)} (P : is_limit c) :
+  is_limit (cone_of_cone_uncurry Q c) :=
+{ lift := λ s, P.lift
+  { X := s.X,
+    π :=
+    { app := λ p, s.π.app p.1 ≫ (D.obj p.1).π.app p.2,
+      naturality' := λ p p' f,
+      begin
+        dsimp, simp only [category.id_comp, category.assoc],
+        rcases p with ⟨j, k⟩,
+        rcases p' with ⟨j', k'⟩,
+        rcases f with ⟨fj, fk⟩,
+        dsimp,
+        slice_rhs 3 4 { rw ←nat_trans.naturality, },
+        slice_rhs 2 3 { rw ←(D.obj j).π.naturality, },
+        simp only [functor.const.obj_map, category.id_comp, category.assoc],
+        have w := (D.map fj).w k',
+        dsimp at w,
+        rw ←w,
+        have n := s.π.naturality fj,
+        dsimp at n,
+        simp only [category.id_comp] at n,
+        rw n,
+        simp,
+      end, } },
+  fac' := λ s j,
+  begin
+    apply (Q j).hom_ext,
+    intro k,
+    simp,
+  end,
+  uniq' := λ s m w,
+  begin
+    refine P.uniq { X := s.X, π := _, } m _,
+    rintro ⟨j, k⟩,
+    dsimp,
+    rw [←w j],
+    simp,
+  end, }
+
+section
+variables (F)
+variables [has_limits_of_shape K C]
+
+/--
+Given a functor `F : J ⥤ K ⥤ C`, with all needed chosen limits,
+we can construct a diagram consisting of the limit cone over each functor `F.obj j`,
+and the universal cone morphisms between these.
+-/
+@[simps]
+def diagram_of_cones.mk_of_has_limits : diagram_of_cones F :=
+{ obj := λ j, limit.cone (F.obj j),
+  map := λ j j' f, { hom := lim.map (F.map f), }, }
+
+-- Satisfying the inhabited linter.
+instance diagram_of_cones_inhabited : inhabited (diagram_of_cones F) :=
+⟨diagram_of_cones.mk_of_has_limits F⟩
+
+@[simp]
+lemma diagram_of_cones.mk_of_has_limits_cone_points :
+  (diagram_of_cones.mk_of_has_limits F).cone_points = (F ⋙ lim) :=
+rfl
+
+variables [has_limit (uncurry.obj F)]
+variables [has_limit (F ⋙ lim)]
+
+/--
+The Fubini theorem for a functor `F : J ⥤ K ⥤ C`,
+showing that the limit of `uncurry.obj F` can be computed as
+the limit of the limits of the functors `F.obj j`.
+-/
+def fubini : limit (uncurry.obj F) ≅ limit (F ⋙ lim) :=
```

Comment:
I think it's great that this has `Fubini` in the docstring, but I would nevertheless suggest that the actual name be something like `limit_uncurry_iso_limit_comp_lim` (feel free to choose something else).
```suggestion
def limit_uncurry_iso_limit_comp_lim : limit (uncurry.obj F) ≅ limit (F ⋙ lim) :=
```